### PR TITLE
Adds Boot Knives (Combat + Mining Boots + Jump Boots)

### PR DESCRIPTION
**Pull request by matttheficus**
## What Does This PR Do
Allows you to place a Survival or Combat knife into either Mining, Jump or Combat boots. You can draw it by alt-clicking the boot.

Adds Combat Boots to the SWAT Crate and increases its price from `800` to `900` since this allows for easy knife carrying.

## Why It's Good For The Game
Cool and thematic. Also, some servers already have this and I think its neat. Having a holdout knife in a fight is also useful.

## Testing
Stabbed myself in the toes a few times before I got it right.

## Changelog
:cl:
add: Combat, Jump and Mining Boots can now hold survival/combat knives.
tweak: SWAT Crate price changed from 800 to 900 due to slight buff.
/:cl:

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -521,6 +524,23 @@
...
 				to_chat(user, "<span class='notice'>[src] have already had [p_their()] toes cut open!</span>")
 		return
+
+	if(istype(I, /obj/item/kitchen/knife/combat))
+		var/obj/item/kitchen/knife/combat/W = I
+		if(!knife_slot)
+			to_chat(user, "There is no place to put [W] in [src]!")
+			return
+		if(hidden_blade)
+			to_chat(user, "There is already something in [src]!")
```
Comment (Henri215) [inline code/modules/clothing/clothing.dm:534]: could use a span class

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -542,6 +562,32 @@
...
+			"<span class='notice'>You draw [hidden_blade] from your [name]!</span>")
+		user.put_in_hands(hidden_blade)
+		hidden_blade.add_fingerprint(user)
+		hidden_blade = null
+
+/obj/item/clothing/shoes/examine(mob/user)
+	. = ..()
+
+	if(knife_slot)
+	. += "<span class='info'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src]</span>"
```
Comment (Henri215) [inline code/modules/clothing/clothing.dm:589]:
```suggestion
	. += "<span class='info'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src].</span>"
```

Comment (matttheficus) [reply code/modules/clothing/clothing.dm]: Just caught this one, last second change momento.

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -521,6 +524,23 @@
...
+			to_chat(user, "<span class='notice'>There is no place to put [W] in [src]!</span>")
+			return
+		if(hidden_blade)
+			to_chat(user, "<span class='notice'>There is already something in [src]!</span>")
+			return
+		if(!user.unEquip(W))
+			return
+		user.visible_message("<span class='notice'>[user] places [W] into their [name]!</span>", \
+			"<span class='notice'>You place [W] into the side of your [name]!</span>")
+		W.loc = src
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:540]:
```suggestion
		W.forcemove(src)
```
image

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -542,6 +562,32 @@
 	icon_state = "[icon_state]_opentoe"
 	item_state = "[item_state]_opentoe"
 
+/obj/item/clothing/shoes/AltClick(mob/user)
+	if(user.stat || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || !Adjacent(user))
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:566]: Stat is implied by TRAIT_HANDS_BLOCKED
```suggestion
	if(HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || !Adjacent(user))
```

File: code/modules/clothing/clothing.dm [hunk 5]
```diff
@@ -542,6 +562,32 @@
...
+		return
+
+	if(!knife_slot)
+		return
+	if(!hidden_blade)
+		to_chat(user, "<span class='warning'>There's nothing in your [name]!</span>")
+		return
+
+	if(isobj(user.get_active_hand()) && isobj(user.get_inactive_hand()))
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:575]:
```suggestion
	if(!isnull(user.get_active_hand()) && !isnull(user.get_inactive_hand()))
```

File: code/modules/clothing/clothing.dm [hunk 6]
```diff
@@ -542,6 +562,32 @@
...
+		to_chat(user, "<span class='warning'>You need an empty hand to pull [hidden_blade]!</span>")
+		return
+	else
+		user.visible_message("<span class='notice'>[user] pulls [hidden_blade] from their [name]!</span>", \
+			"<span class='notice'>You draw [hidden_blade] from your [name]!</span>")
+		user.put_in_hands(hidden_blade)
+		hidden_blade.add_fingerprint(user)
+		hidden_blade = null
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:583]:
```suggestion
	user.visible_message("<span class='notice'>[user] pulls [hidden_blade] from their [name]!</span>", \
		"<span class='notice'>You draw [hidden_blade] from your [name]!</span>")
	user.put_in_hands(hidden_blade)
	hidden_blade.add_fingerprint(user)
	hidden_blade = null
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/clothing/clothing.dm [hunk 7]
```diff
@@ -521,6 +524,23 @@
 			else
 				to_chat(user, "<span class='notice'>[src] have already had [p_their()] toes cut open!</span>")
 		return
+
+	if(istype(I, /obj/item/kitchen/knife/combat))
+		var/obj/item/kitchen/knife/combat/W = I
```
Comment (GDNgit) [inline code/modules/clothing/clothing.dm:529]:
```suggestion
		var/obj/item/kitchen/knife/combat/W = I
```
Is this cast needed

File: code/modules/clothing/clothing.dm [hunk 8]
```diff
@@ -521,6 +524,23 @@
...
+			to_chat(user, "<span class='notice'>There is already something in [src]!</span>")
+			return
+		if(!user.unEquip(W))
+			return
+		user.visible_message("<span class='notice'>[user] places [W] into their [name]!</span>", \
+			"<span class='notice'>You place [W] into the side of your [name]!</span>")
+		W.forceMove(src)
+		hidden_blade = W
+
+
```
Comment (GDNgit) [inline code/modules/clothing/clothing.dm:543]:
```suggestion
```

Comment (Henri215) [inline code/modules/clothing/clothing.dm:543]:
```suggestion
```

File: code/modules/clothing/clothing.dm [hunk 9]
```diff
@@ -542,6 +562,32 @@
...
+	user.visible_message("<span class='notice'>[user] pulls [hidden_blade] from their [name]!</span>", \
+		"<span class='notice'>You draw [hidden_blade] from your [name]!</span>")
+	user.put_in_hands(hidden_blade)
+	hidden_blade.add_fingerprint(user)
+	hidden_blade = null
+
+/obj/item/clothing/shoes/examine(mob/user)
+	. = ..()
+
```
Comment (GDNgit) [inline code/modules/clothing/clothing.dm:587]:
```suggestion
```

File: code/modules/clothing/clothing.dm [hunk 10]
```diff
@@ -542,6 +562,32 @@
...
+	if(knife_slot)
+		. += "<span class='info'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src]</span>"
```
Comment (GDNgit) [inline code/modules/clothing/clothing.dm:589]: Make this tell you if you have a knife or not rather than making people guess

Comment (Henri215) [inline code/modules/clothing/clothing.dm:589]:
```suggestion
		. += "<span class='info'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src].</span>"
```

File: code/modules/clothing/clothing.dm [hunk 11]
```diff
@@ -542,6 +562,32 @@
 	icon_state = "[icon_state]_opentoe"
 	item_state = "[item_state]_opentoe"
 
+/obj/item/clothing/shoes/AltClick(mob/user)
+	if(HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || !Adjacent(user))
+		return
+
+	if(!knife_slot)
+		return
```
Comment (GDNgit) [inline code/modules/clothing/clothing.dm:570]:
```suggestion
	if(HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || !Adjacent(user) || !knife_slot)
```

File: code/modules/clothing/clothing.dm [hunk 12]
```diff
@@ -542,6 +562,32 @@
...
+	if(!hidden_blade)
+		to_chat(user, "<span class='warning'>There's nothing in your [name]!</span>")
+		return
+
+	if(!isnull(user.get_active_hand()) && !isnull(user.get_inactive_hand()))
```
Comment (GDNgit) [inline code/modules/clothing/clothing.dm:575]:
```suggestion
	if(user.get_active_hand() && user.get_inactive_hand())
```
Just check if there's a value here, no need for a double not operator. this also really shouldn't be hand dependent, just force it on the floor if their hands are full

File: code/modules/clothing/clothing.dm [hunk 13]
```diff
@@ -542,6 +562,32 @@
...
+		to_chat(user, "<span class='warning'>You need an empty hand to pull [hidden_blade]!</span>")
```
Comment (GDNgit) [inline code/modules/clothing/clothing.dm:576]:
```suggestion
		to_chat(user, "<span class='warning'>You need an empty hand to pull out [hidden_blade]!</span>")
```

**Review by GDNgit — CHANGES_REQUESTED** [review]

File: code/modules/clothing/clothing.dm (hunk 12 above)
Comment (matttheficus) [reply code/modules/clothing/clothing.dm:575]: I'm copying the same drawing logic from Holsters, which also require an open hand. Making a dexterous movement like drawing a weapon should require an open hand - I wont be budging on that one.

File: code/modules/clothing/clothing.dm (hunk 7 above)
Comment (matttheficus) [reply code/modules/clothing/clothing.dm:529]: It seems not - copy/paste code logic moment :gatto:

File: code/modules/clothing/shoes/misc_shoes.dm [hunk 14]
```diff
@@ -411,6 +413,7 @@
 	actions_types = list(/datum/action/item_action/bhop)
 	permeability_coefficient = 0.05
 	can_cut_open = FALSE
+	knife_slot = TRUE
 	var/jumpdistance = 5 //-1 from to see the actual distance, e.g 4 goes over 3 tiles
 	var/jumpspeed = 3
 	var/recharging_rate = 60 //default 6 seconds between each dash
```
Comment (Burzah) [inline code/modules/clothing/shoes/misc_shoes.dm:419]:
```suggestion
	var/recharging_rate = 6 SECONDS //default time between each dash
```
I know you didn't touch it, but I can't unsee it.

**Review by Burzah — CHANGES_REQUESTED** [review]

File: code/modules/clothing/clothing.dm [hunk 15]
```diff
@@ -536,6 +553,30 @@
...
+	user.put_in_hands(hidden_blade)
+	hidden_blade.add_fingerprint(user)
+	hidden_blade = null
+
+/obj/item/clothing/shoes/examine(mob/user)
+	. = ..()
+	if(knife_slot)
+		. += "<span class='info'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src].</span>"
+		if(hidden_blade)
+			. += "<span class='info'>Your boot has a [hidden_blade.name] hidden inside of it!</span>"
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:578]: Any reason to use `info` here instead of `notice`?

File: code/modules/clothing/clothing.dm [hunk 16]
```diff
@@ -515,6 +518,20 @@
...
+			to_chat(user, "<span class='notice'>There is already something in [src]!</span>")
+			return
+		if(!user.unEquip(I))
+			return
+		user.visible_message("<span class='notice'>[user] places [I] into their [name]!</span>", \
+			"<span class='notice'>You place [I] into the side of your [name]!</span>")
+		I.forceMove(src)
+		hidden_blade = I
 	else
 		return ..()
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:536]:
```suggestion
		hidden_blade = I
		return
	return ..()
```

File: code/modules/clothing/clothing.dm [hunk 17]
```diff
@@ -462,6 +462,9 @@
 	var/can_cut_open = FALSE
 	var/cut_open = FALSE
 	var/no_slip = FALSE
+	var/knife_slot = FALSE
+	var/obj/item/kitchen/knife/combat/hidden_blade = null
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:466]:
```suggestion
	var/obj/item/kitchen/knife/combat/hidden_blade
```

**Review by Contrabang — CHANGES_REQUESTED** [review]
Some minor stuff

File: code/modules/clothing/shoes/misc_shoes.dm (hunk 14 above)
Comment (matttheficus) [reply code/modules/clothing/shoes/misc_shoes.dm:419]: Nope! Swapped.

File: code/modules/clothing/clothing.dm (hunk 15 above)
Comment (matttheficus) [reply code/modules/clothing/clothing.dm:578]: Nope swapped!

File: code/modules/clothing/clothing.dm [hunk 18]
```diff
@@ -536,6 +554,30 @@
...
+	user.visible_message("<span class='notice'>[user] pulls [hidden_blade] from their [name]!</span>", \
+		"<span class='notice'>You draw [hidden_blade] from your [name]!</span>")
+	user.put_in_hands(hidden_blade)
+	hidden_blade.add_fingerprint(user)
+	hidden_blade = null
+
+/obj/item/clothing/shoes/examine(mob/user)
+	. = ..()
+	if(knife_slot)
+		. += "<span class='info'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src].</span>"
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:577]:
```suggestion
		. += "<span class='notice'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src].</span>"
```

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Review by Contrabang — APPROVED** [review]
LGTM

**Pull request merged**